src/main/java/edu/umich/its/cpm/MigrationInstanceService.java

```diff
@@ -38,13 +39,45 @@
 	private MigrationBoxFileRepository fRepository;
 	
 	@Autowired
+	private MigrationEmailMessageRepository eRepository;
+	
+	@Autowired
 	private MigrationRepository mRepository;
 	
+	@Autowired
+	private Environment env;
+	
+	/**
+	 * check the setting of parallel processing thread number
+	 * @return
+	 */
+	private int getMaxParallelThreadNum()
+	{
+		if (env.getProperty(Utils.MAX_PARALLEL_THREADS_PROP) != null)
+		{
+			try
+			{
+				// return the setting in property file
+				return Integer.parseInt(env.getProperty(Utils.MAX_PARALLEL_THREADS_PROP));
+			}
+			catch (NumberFormatException e)
+			{
+				// log error and return default value
+				log.error(Utils.MAX_PARALLEL_THREADS_PROP + " property should have integer value. ");
+				return Utils.MAX_PARALLEL_THREADS_NUM;
```

Comment:
I'd argue it shouldn't start up.  If the number is invalid why would we think that everything isn't screwed up?